<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand -->
<html lang="en">

<head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Tools for analyzing TCR-clonotype data • clonotypr</title>
  <!-- jquery -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js"
    integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap -->
  <link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet"
    crossorigin="anonymous">
  <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js"
    integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>
  <!-- bootstrap-toc -->
  <link rel="stylesheet" href="bootstrap-toc.css">
  <script src="bootstrap-toc.js"></script><!-- Font Awesome icons -->
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css"
    integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css"
    integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
  <!-- clipboard.js -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js"
    integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>
  <!-- headroom.js -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js"
    integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js"
    integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown -->
  <link href="pkgdown.css" rel="stylesheet">
  <script src="pkgdown.js"></script>
  <meta property="og:title" content="Tools for analyzing TCR-clonotype data">
  <meta property="og:description" content="Clonotypr offers several functions for 
    analyzing TCR-clonotype data. One of the primary benefits
    is the integration of clonotype data into Seurat 
    objects (https://satijalab.org/seurat/) as metadata.
    The main tools include functions for: attaching 
    clonotype data, clonotype descriptives, and 
    isolating clonotypes based on Seurat object 
    params (i.e gene expression matrices). 
    Clonotypr was developed using example 
    scRNA/TCR-seq data generetd by the CellRanger
    pipeline from the 10X Genomics platform 
    (https://www.10xgenomics.com/). Support for
    other data types is currently in progress. Please
    contact the developer for updates.">
  <!-- mathjax -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js"
    integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js"
    integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>
  <!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-137030283-2"></script>
  <script>
    window.dataLayer = window.dataLayer || [];

    function gtag() {
      dataLayer.push(arguments);
    }
    gtag('js', new Date());

    gtag('config', 'UA-137030283-2');
  </script>
</head>

<body data-spy="scroll" data-target="#toc">
  <div class="container template-home">

    {% include 'clonotypr/navbar.html' %}

    <div class="row">
      <div class="contents col-md-9">

        <p>
          <!-- <img src=" {{ url_for('static', filename='clonotypr/logo.png') }}" alt="clonotypr logo"> -->
        </p>
        
        
        <div id="clonotypr" class="section level1">
          <div class="page-header">
            <div class="alert text-dark text-center" style='background-color:cornsilk;'>
              Clonotypr is being released as part of the upcoming <a href='https://www.labvivo.com'>LabVivo</a> release
            </div>
            <h1 class="hasAnchor">
              <a href="#clonotypr" class="anchor"></a><strong>clonotypr</strong>
            </h1>
          </div>
          <p><code>clonotypr</code> is an R package designed to integrate the post-analysis of V(D)J-seq/single-cell
            immune profiling data with scRNA-seq data using the <code>Seurat</code> arsenal as a backbone. Among
            appending clonotype data to objects, clonotypr offers strategies for analyzing CDR3aa lengths, chemical
            composition, and frequency, as well as, reassignment of single-chain clonotypes to their
            ‘productive’-relatives.<br><em>Clonotypr &gt;= 0.2.0 also supports typical flat-file analysis; analysis
              independent of Seurat objects</em></p>
          <p>For any questions, comments or suggestions, please email me at <a href="mailto:matthew.femia@outlook.com"
              class="email">matthew.femia@outlook.com</a></p>
          <p><a href="articles/vignette"
              class="uri">View the tutorial</a></p>
          <div id="clonotypr-is-actively-in-development-to-bring-more-features-and-support-for-other-analysis-pipelines"
            class="section level4">
            <h4 class="hasAnchor">
              <a href="#clonotypr-is-actively-in-development-to-bring-more-features-and-support-for-other-analysis-pipelines"
                class="anchor"></a><code>clonotypr</code> is actively in development to bring more features and support
              for other analysis pipelines</h4>
            <hr>
          </div>
          <div id="installation" class="section level2">
            <h2 class="hasAnchor">
              <a href="#installation" class="anchor"></a>Installation</h2>
            <div class="sourceCode" id="cb1">
              <pre
                class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="st">"devtools"</span>)) {
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
}

<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"mattfemia/clonotypr"</span>)</pre>
            </div>
            <hr>
          </div>
          <div id="package-details" class="section level2">
            <h2 class="hasAnchor">
              <a href="#package-details" class="anchor"></a>Package Details</h2>
            <p>Available functions (0.4.0.9000):</p>
            <p><code>AddClonotypes</code> – Attach clonotype data from sequencing pipeline. v0.1.0 only supports
              CellRanger (10X Genomics)<br><code>AnalyzeCDR3Length</code> – Build a dataframe containing CDR3alpha seqs
              + lengths and CDR3 beta seqs + lengths<br><code>AnalyzeCDR3aa</code> – Calculates position matrices (PFM,
              PPM, or PWM)<br><code>AnalyzeGenes</code> - Build a summary dataframe with V(D)J/Ig gene
              usage<br><code>BuildFM</code>, <code>BuildPPM</code>, <code>BuildPWM</code> – Builds position matrices for
              AA composition in CDR3 alpha or beta chains<br><code>BuildSummary</code> – Merges output from
              <code>MakeFrequencyDF</code> and <code>AnalyzeCDR3Lengths</code> with original clonotypes summary data to
              build new table<br><code>BuildVDJNetwork</code> – Network analysis for VDJ genes across the sample.
              Returns an edge list, node list, and full list of connections between VDJ
              genes<br><code>DescribeClonotypes</code> – Prettyprints/returns a list with the number of productive,
              single-chain, and multi-chain clonotypes<br><code>GetBarcodeFreqs</code> – Formats and merges returns from
              <code>MakeBarcodeDF</code> and <code>MakeFrequencyDF</code><br><code>JoinSingleChains</code> – Integrates
              single-chain clonotypes into productive chain clonotypes and recalculates clone
              frequencies<br><code>LoadClonotypes</code> – Read clonotype data from CellRanger CSVs into a formatted
              dataframe<br><code>MakeBarcodeDF</code> – Pulls barcode and clonotype_id data
              together<br><code>MakeFrequencyDF</code> – Calculates frequency and percent:whole of each unique clonotype
              in population<br><code>PlotCDR3aa</code> – Generates ggseqlogo plots from CDR3 AA composition. Plot for
              both alpha and beta chains<br><code>PlotFrequencyDist</code> – Plots ggplot for clonotype frequency
              ranges<br><code>PlotLengths</code> – Generates ggplots for CDR3 lengths. Plot for both alpha and beta
              chains<br><code>PlotPFM</code>, <code>PlotPPM</code>, <code>PlotPWM</code> – Plots heatmaps for CDR3 amino
              acid position matrices<br><code>PlotTopClonotypes</code> – Plots ggplot for the top n-clonotype
              frequencies<br><code>PlotVDJBySegment</code> – Plots histogram of all genes in a given T-Cell/B-Cell
              chain/isotype for a given VDJ-gene (i.e “TRAV”, “IGKJ”)<br><code>PlotVDJStatus</code> – Plots cell counts
              for the presence versus absence of a given VDJ gene<br><code>PlotVDJByNetwork</code> – Plots sankey
              diagram or chord diagram based on the network analysis done by
              <code>BuildVDJNetwork</code><br><code>WhichClonotypes</code> – Pulls clonotype_ids from Seurat object
              based on gene filtering. Based on Seurat::WhichCells.</p>
          </div>
        </div>

      </div>

      <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
        <div class="links">
          <h2>Links</h2>
          <ul class="list-unstyled">
            <li>Browse source code at <br><a
                href="http://github.com/mattfemia/clonotypr/">http://​github.com/​mattfemia/​clonotypr/​</a>
            </li>
            <li>Report a bug at <br><a
                href="http://github.com/mattfemia/clonotypr/issues">http://​github.com/​mattfemia/​clonotypr/​issues</a>
            </li>
          </ul>
        </div>
        <div class="license">
          <h2>License</h2>
          <ul class="list-unstyled">
            <li>
              <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file <a
                href="/license">LICENSE</a>
            </li>
          </ul>
        </div>
        <div class="developers">
          <h2>Developers</h2>
          <ul class="list-unstyled">
            <li>Matt Femia <br><small class="roles"> Maintainer </small> </li>
          </ul>
        </div>

      </div>
    </div>


    <footer>
      <div class="copyright">
        <p>Developed by Matt Femia.</p>
      </div>

      <div class="pkgdown">
        <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
      </div>

    </footer>
  </div>




</body>

</html>